UnityClient/Assets/Scripts/TowerDefence/RoundsSurvived.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class RoundsSurvived : MonoBehaviour
+{
+    public Text roundsText;
+
+    void OnEnable()
+    {
+        StartCoroutine(AnimateText());
+    }
+
+    IEnumerator AnimateText()
+    {
+        roundsText.text = "0";
+        int round = 0;
+
+        yield return new WaitForSeconds(.7f);
+
+        while (round < PlayerStats.Rounds)
+        {
+            round++;
+            roundsText.text = round.ToString();
+            
+            yield return new WaitForSeconds(.05f);
```

Comment:
making new `WaitForSeconds` like this every time is expensive. Please create and persist the `WaitForSeconds` you require in private fields and simply `yield return` the fields.